provider/src/main/kotlin/org/gradle/kotlin/dsl/accessors/CodeGenerator.kt

```diff
@@ -229,28 +229,57 @@ fun configurationAccessorFor(name: String): String? =
              * @see DependencyHandler.add
              */
             inline
-            fun <T : ModuleDependency> DependencyHandler.`$name`(dependency: T, dependencyConfiguration: T.() -> Unit): T =
-                add("$name", dependency, dependencyConfiguration)
+            fun <T : ModuleDependency> DependencyHandler.`${name.kotlinIdentifier}`(dependency: T, dependencyConfiguration: T.() -> Unit): T =
+                add("${name.stringLiteral}", dependency, dependencyConfiguration)
 
         """
     }
 
 
 private
-fun documentInaccessibilityReasons(name: String, typeAccess: TypeAccessibility.Inaccessible): String =
-    "`$name` is not accessible in a type safe way because:\n${typeAccess.reasons.map { reason ->
+data class AccessorNameSpec(val original: String,
+                            val kotlinIdentifier: String = original,
+                            val stringLiteral: String = stringLiteralFor(original))
```

Comment:
:x: Considering that these specs can be filtered and never used I would make this a lazy property:

```kotlin
    val stringLiteral by lazy { stringLiteralFor(original) }
```